core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DatalakeV4Controller.java

```diff
@@ -28,11 +31,46 @@ public class DatalakeV4Controller implements DatalakeV4Endpoint {
     @Inject
     private WorkspaceService workspaceService;
 
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
     @Override
+    @InternalOnly
     public StackViewV4Responses list(@Deprecated String environment, @TenantAwareParam String environmentCrn) {
         if (StringUtils.isBlank(environmentCrn) || StringUtils.isNotBlank(environment)) {
             throw new BadRequestException("environment param is deprecated, only environmentCrn should be filled.");
         }
         return stackOperations.listByEnvironmentCrn(workspaceService.getForCurrentUser().getId(), environmentCrn, List.of(StackType.DATALAKE));
     }
+
+    @Override
+    public Boolean isDiskTypeChangeSupported(String platform) {
+        if (StringUtils.isBlank(platform)) {
+            throw new BadRequestException("Platform param is null or empty.");
+        }
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
+    }
+
+    @Override
+    public void updateDiskTypeAndSize(SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception {
+        DiskUpdateRequest diskUpdateRequest = sdxDiskUpdateRequest.getDiskUpdateRequest();
+        List<VolumeSetAttributes.Volume> volumesToUpdate = sdxDiskUpdateRequest.getVolumesToUpdate();
+        String cloudPlatform = sdxDiskUpdateRequest.getCloudPlatform();
+        diskUpdateService.updateDiskTypeAndSize(diskUpdateRequest, volumesToUpdate, cloudPlatform);
```

Comment:
create service for this logic

Reply:
@horadla23 this is just 3 lines of code. I would have created the service if it is warranted. but for 3 lines of code - which is nothing but getting values from input object it did not seem relevant. it is calling the diskupdateservice with these values. I have made it call diskupdateservice with values specifically and not the entire object, because it is going to be reused for datahubs as well once this is merged. i am not going to create service for this

Reply:
i see, ok, if there is only 3 lines, it will be fine